### ISSUE TITLE
Gradients of determinant for matrices w/corank > 1 on TPU

### ISSUE THREAD

**Issue by dpfau** [google/jax#2852]
The gradient of the determinant op currently fails on TPU for any matrix of rank less than n-1, though it correctly returns a matrix of zeros on CPU and GPU.

**Comment by rajasekharporeddy** [comment]
Hi @dpfau 

I tried to execute an example code to compute the gradient of determinant for matrix of rank less than `n-1` with JAX version of 0.3.25 on Google Colab TPU. It returned a matrix of zeros on TPU also.

```python
import jax.numpy as jnp
from jax import vmap, grad
import jax


x = jax.random.normal(jax.random.PRNGKey(0), (10,10))
x = x.at[5:, :].set(x[0:5,:])
x = x.at[5:, :].multiply(2)
# print(x)
print(f"Shape of the matrix: {x.shape}")
print(f"Rank of the matrix: {jnp.linalg.matrix_rank(x)}")
grad_det = grad(jnp.linalg.det)(x)  # x is your input matrix

grad_det
```

Output:

```
Shape of the matrix: (10, 10)
Rank of the matrix: 6
DeviceArray([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
             [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]], dtype=float32)
```

Kindly find the [gist](https://colab.research.google.com/gist/rajasekharporeddy/8c5e985d04b24755360d642632b1b219/tpu_-2852.ipynb) for reference.

Thank you
